source/unified-test-format/unified-test-format.rst

```diff
@@ -1798,6 +1897,119 @@ Use a ``listIndexes`` command to check whether the index exists. Note that it is
+  arising during sub-operation execution and append a document with failure
+  information to the array stored in the specified entity. If
+  not specified, the runner MUST treat failures as errors, and either
+  handle them following the logic described in ``storeErrorsAsEntity``
+  or cause them to terminate execution, if ``storeErrorsAsEntity`` is not
+  specified. The failure document MUST contain the following fields:
+  
+  - ``error``: the textual description of the failure encountered.
+  - ``time``: the number of (floating-point) seconds since the Unix epoch
+    when the failure was encountered.
+
+- ``storeIterationsAsEntity``: if specfied, the runner MUST keep track of
+  the number of iterations of the loop performed, and store that number
+  in the specified entity.
+
+A *failure* is when an operation performed by the test runner has a
+different result from what the test expected. For example, a test
+containing an expectation that a particular document has a particular value
+in a particular field, and the actual value being different, would
+be a failure. An *error* is any other type of error. For example,
+attempting to invoke an undefined operation would be an error.
+
+This specification permits the test runner to report some failures as errors
+and some errors as failures. When the test runner stores errors and
+failures as entities it MAY classify conditions as errors and failures in
+the same way as it would when used in the driver's test suite.
+This includes reporting all errors as failures or all failures as errors.
+
+If the test runner does not distinguish errors and failures in its reporting,
+it MAY report both conditions under either category, but it MUST report
+any given condition in at most one category.
+
+The same entity name MAY be used for both ``storeErrorsAsEntity``
+and ``storeFailuresAsEntity`` options. The test runner MUST support
+writing both errors and failures to the same entity. If the same entity is
+specified, the errors and failures MUST be written in chronological order
+to the extent possible, interleaving the error and failure entries
+if necessary. Note that the textual description field for both errors
+and failures is ``error``.
+
+The following termination behavior MUST be implemented by the unified test
+runner:
+
+- There MUST be a way to request termination of the loops. This request
+  will be made by the Atlas testing workload executor in response to
+  receiving the termination signal from Astrolabe.
+  
+- When the termination request is received, the workload executor MUST
+  stop looping. The current loop iteration SHOULD complete to its natural
+  conclusion (success or failure).
+
+- After the termination request is received, the runner MUST NOT start
+  any further loops.
+  
+- When the termination request is received, the runner SHOULD skip
+  any non-loop operations not yet started and terminate as soon as practical.
```

Comment:
These last two bullets suggest that the signal isn't merely to stop the loop but rather the entire test. This seems at odds with the first bullet about "request(ing) termination of the loops" and allowing the current iteration to run its course. I propose we delete this.

I'll elaborate on my reasoning below (let me know if you disagree with anything):

 * This spec does not discuss how signals (i.e. requests for termination) are provided to the test runner. They could either be proactively sent from an external source (e.g. IPC) or periodically checked by the test runner (e.g. asking a callback if it's OK to continue).
 * Any request for termination is effectively a request to terminate the current loop being executed (if any). If the test runner is not presently in a loop, the request has no effect (and may very well be ignored entirely).
 * Acknowledging a request for termination should have no impact on future tests or loops. FWIW, I also suggest we prohibit multiple or nested loops in another comment.
 * A request for termination allows the current iteration (if any) to finish executing.

If we allow the current iteration to wrap up, I don't see any problem with allowing the test to finish executing the remainder of its `operations` array. In practice, I don't expect the workload executor will care one way of another, since its tests are likely to consist of a single `loop` operation. Thoughts?